WalletWasabi/Wallets/WalletManager.cs

```diff
@@ -46,10 +62,13 @@ public WalletService GetFirstOrDefaultWallet()
 			}
 		}
 
-		public async Task AddAndStartAsync(WalletService walletService, CancellationToken cancel)
+		public async Task<WalletService> CreateAndStartWalletServiceAsync(KeyManager keyManager)
 		{
+			WalletService walletService = new WalletService(BitcoinStore, keyManager, Synchronizer, Nodes, DataDir, ServiceConfiguration, FeeProvider, BitcoinCoreNode);
+			using (CancelWalletServiceInitialization = new CancellationTokenSource())
```

Comment:
This should be a per wallet lock. In this way, it makes no sense for me to define it here:

- It is not thread-safe. 
- If another wallet creation is in progress, meanwhile run this again, the reference will be lost as a result it won't be canceled. 

You can have a `CancellationTokenSource` property in WalletManager which is there from the beginning and canceling all wallet creation on `RemoveAndStopAllAsync`.
Any other solution will require to have per wallet lock and the touching of `WalletService`.

Reply:
Yeah @danwalmsley should've pick up this work yesterday and continue it with this. Not sure what happened.

Although that's a feature request, so I'm ok with not adding that feature to this PR, too.

Reply:
Working on it ;)

Reply:
Make sure to pull, I addressed @molnard's other comments.

Reply:
👍 iv pushed, what do you think of my WalletState approach?

Reply:
For me, it is a Concept NACK. The state of the object should be added to the object itself, supplementary state handling causes maintenance overhead and very error-prone. In this case, this should be in WalletService.